src/Microsoft.VisualStudio.Threading/TplExtensions.cs

```diff
@@ -53,7 +53,7 @@ public static void WaitWithoutInlining(this Task task)
             if (!task.IsCompleted)
             {
                 // Waiting on a continuation of a task won't ever inline the predecessor (in .NET 4.x anyway).
-                var continuation = task.ContinueWith(t => { }, CancellationToken.None, TaskContinuationOptions.None, TaskScheduler.Default);
+                var continuation = task.ContinueWith(t => { }, CancellationToken.None, TaskContinuationOptions.RunContinuationsAsynchronously, TaskScheduler.Default);
```

Comment:
We never expose this continuation and no continuations are ever scheduled on it. Why would this flag have any impact on anything?

Reply:
So far I have not been able to figure this out.

Reply:
well, maybe my comment that this works "in .NET 4.x anyway" isn't true any more. Maybe it only worked on .NET 4.0.